forte/processors/base/data_augment_processor.py

```diff
@@ -0,0 +1,115 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processors that augment the data. The processor will call
+replacement ops to generate texts similar to those in the input pack
+and create a new pack with them.
+"""
+from typing import List, Tuple
+from forte.data.ontology.core import Entry
+from forte.data.ontology.top import Annotation
+from forte.data.data_pack import DataPack
+from forte.processors.base.base_processor import BaseProcessor
+from forte.processors.data_augment.algorithms.text_replacement_op \
+    import TextReplacementOp
+
+__all__ = [
+    "BaseDataAugmentProcessor",
+    "ReplacementDataAugmentProcessor"
+]
+
+
+class BaseDataAugmentProcessor(BaseProcessor):
+    r"""The base class of processors that augment data.
+    This processor instantiates replacement ops where specific
+    data augmentation algorithms are implemented. The replacement ops
+    will run the algorithms and the processor will create Forte
+    data structures based on the augmented inputs.
+    """
+
+
+class ReplacementDataAugmentProcessor(BaseDataAugmentProcessor):
+    r"""
+    Most of the Data Augmentation(DA) methods can be
+    considered as replacement-based methods with different
+    levels: character, word, sentence or document.
+    """
+    def __init__(self):
+        """
+        The replaced_spans records the entries replaced by new texts.
+        It consists of tuples (entry, new text) inserted by :func: replace.
+        The new text will be used for building new data pack.
+        """
+        super().__init__()
+        self.replaced_spans: List[Tuple[Entry, str]] = []
+
+    def replace(self, replacement_op: TextReplacementOp, input: Entry):
+        """
+        This is a wrapper function to call the replacement op. After
+        getting the augmented text, it will register the input & output
+        for later batch process of building the new data pack.
+        """
+        replaced_text: str = replacement_op.replace(input)
+        self.replaced_spans.append((input, replaced_text))
+
+    def auto_align_annotations(
+        self,
+        data_pack: DataPack,
+        replaced_annotations: List[Tuple[Annotation, str]]
+    ) -> DataPack:
+        r"""
+        Function to replace some annotations with new strings.
+        It will update the text and auto-align the annotation spans.
+
+        Args:
+            data_pack: Datapack holding the annotations to be replaced.
+            replaced_annotations: A list of tuples(annotation, new string).
+            The text for annotation will be updated with the new string.
+
+        Returns:
+            A new data_pack holds the text after replacement. The annotations
+            in the original data pack will be copied and auto-aligned as
+            instructed by the "other_entry_policy".
+
+        """
```

Comment:
Oh, can you do this? Do you need to put a pass keyword here?

Reply:
What is a pass keyword? I thought we can just access the config and get the "other_entry_policy" from it.

Reply:
No, I am just wondering if  it is OK to leave the function empty. Normally I will use the `pass` statement: https://docs.python.org/2.0/ref/pass.html

Reply:
Yeah, I just have a try, so having the docstring will serve the purpose, we don't need the pass statement.